compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1287,6 +1287,11 @@ public void visit(BLangErrorType errorTypeNode) {
         BType detailType = Optional.ofNullable(errorTypeNode.detailType)
                 .map(bLangType -> resolveTypeNode(bLangType, env)).orElse(symTable.detailType);
 
+        if (errorTypeNode.isAnonymous) {
+            errorTypeNode.flagSet.add(Flag.PUBLIC);
```

Comment:
Do we need PUBLIC also?

Reply:
We usually do this to handle the case where this local definition could be a part of a public construct.
`public function foo() returns error<{ int code; }> { }`